Mathlib/Data/Polynomial/Module.lean

```diff
@@ -350,3 +352,79 @@ theorem comp_smul (p p' : R[X]) (q : PolynomialModule R M) :
 #align polynomial_module.comp_smul PolynomialModule.comp_smul
 
 end PolynomialModule
+
+end CommSemiring
+
+namespace Module
+
+open Polynomial
+/--
+Suppose `a` is an element of an `R`-algebra `A` and `M` is an `A`-module.
+Then `Module.AEval R M a` is the `R[X]`-module with carrier `M`,
+where the action of `f : R[X]` is `f • m = (aeval a f) • m`.
+-/
+structure AEval (R M: Type*) {A : Type*} [CommSemiring R] [Semiring A] [Algebra R A]
```

Comment:
I don't think there is any need to create a new structure here, you can just use a type synonym. In practice you define `AEval R M a := M` (as a type) and then you provide the `R`-module structure by hand. You can have a look of what we do for [PiLp](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/NormedSpace/PiLp.html#PiLp), do define the product with the Lp metric.

Reply:
Thank you for the really quick response.

In my original PR on this (7419) I did define the module as a type synonym, and was advised that it would be clearer to make it a single-field structure. I will change it back.

Reply:
Well, I am not 100% sure. @eric-wieser what do you think?